event/events.go

```diff
@@ -21,20 +21,27 @@ import (
 	"encoding/hex"
 	"strings"
 
-	evts "github.com/tendermint/go-events"
+	"fmt"
+
+	"github.com/eris-ltd/eris-db/txs"
+	go_events "github.com/tendermint/go-events"
+	tm_types "github.com/tendermint/tendermint/types"
 )
 
 // TODO improve
 // TODO: [ben] yes please ^^^
 // [ben] To improve this we will switch out go-events with eris-db/event so
 // that there is no need anymore for this poor wrapper.
 
+// Oh for a sum type
```

Comment:
I need to accept `interface{}` because that is what a `TMEventData` (https://github.com/tendermint/tendermint/blob/master/types/events.go#L36), via go-events `EventData` (https://github.com/tendermint/go-events/blob/master/events.go#L11).

So creating a named type that is an `interface{}` is just an idiom that documents what I am expecting here.